### PR TITLE
Added client-oauth2

### PR DESCRIPTION
**Pull request by kryswisnaskas**
**Description**
Added `client-oauth2` for the oauth2 integration. 

**How to test**
- yarn deps
- yarn server
- navigate to `localhost:8080`
- note that the HSES login screen should be presented the first time
- log in
- the hello from ttdap message should display
- refresh the page
- this time there shouldn't be a need to login to see the text

Documentation updated
- [ ] API methods
- [ ] Boundary and Data Flow Diagrams
- [ ] Architectural Decision Records written for major infrastructure decisions with the Nygard template
- [ ] OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -23,4 +23,6 @@ services:
       - "./frontend:/app:rw"
       - "./scripts:/app/scripts"
     environment:
-      - BACKEND_PROXY=http://backend:3001
+      - JEST_JUNIT_OUTPUT_DIR=reports
+      - JEST_JUNIT_OUTPUT_NAME=unit.xml
```
Comment (jasalisbury) [inline docker-compose.yml:27]: These probably got added back in during a merge but `JEST_JUNIT*` vars can be removed from this file.

**Comment by jasalisbury** [comment]
Looks good to me! I'm fine `axios`, could see parallel requests being more helpful on the frontend then the backend. How are we planning on persisting the oauth token and associating it with a frontend user, `express-session`?

**Comment by kryswisnaskas** [comment]
Yes, I was thinking about using `express-session`

File: docker-compose.yml (hunk 1 above)
Comment (kryswisnaskas) [reply docker-compose.yml]: Removed

File: src/middleware/authMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import ClientOAuth2 from 'client-oauth2';
+
+// TODO: move values to env variables
+export const hsesAuth = new ClientOAuth2({
+  clientId: '7b9268101989e6ce7efc',
+  clientSecret: 'ABHnjtF6iB1fsg940xBHvZy89Las7X31gywtsMChHrMV',
```
Comment (jasalisbury) [inline src/middleware/authMiddleware.js:6]: Should these be passed in as env variables?

File: src/middleware/authMiddleware.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * uid that gets set after successful login via HSES. Non-authenticated
+ * users who attempt to access a page that requires authentication will be
+ * redirected to the HSES login page.
+ * @param {*} req - request
+ * @param {*} res - response
+ * @param {*} next - next middleware
+ */
+
+export default async function authMiddleware(req, res, next) {
+  req.session.originalUrl = req.originalUrl;
```
Comment (jasalisbury) [inline src/middleware/authMiddleware.js:26]: Do we need to save the original url here?

File: src/middleware/authMiddleware.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  if (!req.session.userId) {
+    req.session.originalUrl = req.originalUrl; // save the url
+    const uri = hsesAuth.code.getUri();
```
Comment (jasalisbury) [inline src/middleware/authMiddleware.js:29]: Do we need to verify the token is valid and unexpired at this point?

File: src/middleware/authMiddleware.js (hunk 2 above)
Comment (kryswisnaskas) [reply src/middleware/authMiddleware.js:6]: Yes, that's the plan. There is a 'TODO' on line 3.

File: src/middleware/authMiddleware.js (hunk 3 above)
Comment (kryswisnaskas) [reply src/middleware/authMiddleware.js:26]: Good catch. I will remove the redundant assignment.

File: src/middleware/authMiddleware.js (hunk 4 above)
Comment (kryswisnaskas) [reply src/middleware/authMiddleware.js:29]: Yes, we will need that logic. I was going to defer the implementation of it until we have the database set up.

File: src/middleware/authMiddleware.js (hunk 2 above)
Comment (jasalisbury) [reply src/middleware/authMiddleware.js:6]: Oops missed that!

**Pull request merged**